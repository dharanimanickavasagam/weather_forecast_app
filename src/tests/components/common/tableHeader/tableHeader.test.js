import React from "react";
import { shallow } from "enzyme";
import TableHeader from "../../../../components/common/tableHeader";

let mockColumns, wrapper;

beforeEach(() => {
  mockColumns = ["Pressure", "Humidity", "Min °F", "Max °F"];
  wrapper = shallow(<TableHeader columns={mockColumns} />);
});

describe("it renders component TableHeader", () => {
  test("render the table data with the passed data props", () => {
    expect(wrapper).toMatchSnapshot();

    const thead = wrapper.find("thead");
    expect(thead).toHaveLength(1);

    const headers = thead.find("th");
    expect(headers).toHaveLength(mockColumns.length);

    headers.forEach((th, idx) => {
      expect(th.text()).toEqual(mockColumns[idx]);
    });
  });
});
