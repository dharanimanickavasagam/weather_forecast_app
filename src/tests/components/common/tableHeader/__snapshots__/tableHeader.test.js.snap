// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component TableHeader render the table data with the passed data props 1`] = `
<thead>
  <tr>
    <th
      key="0"
      scope="col"
    >
      Pressure
    </th>
    <th
      key="1"
      scope="col"
    >
      Humidity
    </th>
    <th
      key="2"
      scope="col"
    >
      Min °F
    </th>
    <th
      key="3"
      scope="col"
    >
      Max °F
    </th>
  </tr>
</thead>
`;
